src/prices/prices.service.ts

```diff
@@ -0,0 +1,66 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { PRICES_API } from './constants/prices.constants';
+import { AxiosResponse } from 'axios';
+import { firstValueFrom } from 'rxjs';
+import { PricesResponse } from './models/prices.model';
+import { PRICES_ERROR } from './errors/errors';
+import * as dayjs from 'dayjs';
+
+@Injectable()
+export class PricesService {
+  constructor(private readonly httpService: HttpService) {}
+
+  public async getCurrentPrices(): Promise<PricesResponse> {
+    const convertedDate: string = dayjs().format('YYYY-MM-DD');
+    const url: string = `${
+      PRICES_API.PRICES_DEFAULT
+    }${PRICES_API.REPORTS.CURRENT_PRICES(convertedDate)}`;
+
+    try {
+      const response: AxiosResponse<PricesResponse> = await firstValueFrom(
+        this.httpService.get<PricesResponse>(url),
+      );
+      return response.data;
```

Comment:
hmm, I'd perhaps move this part to e.g. `PricesApiService` - there defined basic get / post  etc. methods and included base url there (so `PRICES_API.PRICES_DEFAULT`). And also returned `response.data`. So after this here in `PricesService.getCurrentPrices()` you would call something like this:

```typescript
const response: PricesResponse = await this.pricesApiService.get<PricesResponse>(url);
```

I'm not so sure about constructing a query like this in `PRICES_API.REPORTS.CURRENT_PRICES(convertedDate)` - axios (httpService) should accept query parameters like this:

```typescrupt
httpService.get(url, { params: { $filter: `doba eq '${convertedDate}'` } });
```